elasticsearch_dsl/search.py

```diff
@@ -799,7 +801,7 @@ def execute(self, ignore_cache=False, raise_on_error=True):
             for s, r in zip(self._searches, responses["responses"]):
                 if r.get("error", False):
                     if raise_on_error:
-                        raise TransportError("N/A", r["error"]["type"], r["error"])
+                        raise ApiError("N/A", meta=responses.meta, body=r)
```

Comment:
elasticsearch-py 8.x has two types of errors now, `ApiError` being used for errors reported by Elasticsearch itself and `TransportError` being used for errors at the transport layer (such as connection issues).